### PR TITLE
integration tests for redeem flow

### PR DESCRIPTION
**Pull request by puneet-khushwani-eth**
Creating a draft PR to get some initial feedback on the refactoring of integration tests

### CODE REVIEW

**Comment by deepesh-kn** [comment]
The approach looks good to me 👍

**Comment by abhayks1** [comment]
Reviewing the PR 🐰

File: test_integration/devchains.sh [hunk 1]
```diff
@@ -8,8 +8,13 @@ function start_chains {
 echo starting $originChain chain
 ./node_modules/.bin/mosaic start $originChain
 
+sleep 20
```
[the remark this answers is not in the document]
Comment (puneet-khushwani-eth) [reply test_integration/devchains.sh]: This certainly helps to stabilize resource utilisation.

File: test_integration/mosaic.json [hunk 2]
```diff
@@ -30,7 +30,7 @@
           "gatewayLibAddress": "0x3d0c80965628a7e44AD70E0490DBB4D305F1636D",
           "messageBusAddress": "0xF9234855d1Fb05f06A09baFB827faD084516b21c",
           "merklePatriciaLibAddress": "0x3CaB7dE8Dde1fb082A315F986B09c431c1B7D9e7",
-          "redeemPoolAddress": "0x4c8ba8caecb60c67d69605a772ae1bb9a732fb39"
+          "redeemPoolAddress": "0x8bA9C19BeacBB3eF85E1Df57ceef1Df922F2D87F"
```
[the remark this answers is not in the document]
Comment (puneet-khushwani-eth) [reply test_integration/mosaic.json]: done

File: test_integration/01_init/02_post_init.test.ts [hunk 3]
```diff
@@ -0,0 +1,141 @@
+import * as path from 'path';
```
[the remark this answers is not in the document]
Comment (puneet-khushwani-eth) [reply test_integration/01_init/02_post_init.test.ts]: it also verifies whitelisting of workers thus a generic name

File: test_integration/01_init/02_post_init.test.ts [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+import MosaicConfig from '@openst/mosaic-chains/lib/src/Config/MosaicConfig';
+import { Organization } from '@openst/mosaic-contracts/dist/interacts/Organization';
+import { TransactionReceipt } from 'web3-core';
+import { FacilitatorConfig } from '../../src/Config/Config';
+import Utils from '../Utils';
+import * as Constants from '../Constants.json';
+import assert from '../../test/test_utils/assert';
+import SharedStorage from '../SharedStorage';
+
+describe('facilitator post init', async (): Promise<void> => {
```
[the remark this answers is not in the document]
Comment (puneet-khushwani-eth) [reply test_integration/01_init/02_post_init.test.ts]: it also verifies whitelisting of workers thus a generic name

File: test_integration/01_init/02_post_init.test.ts [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+    );
+    await assertWorkerWhitelisting(
+      txReceipt,
+      organizationInstance,
+      auxiliaryWorker,
+      Constants.auxiliaryWorkerExpirationHeight,
+    );
+  });
+
+  it('funding origin and aux workers', async (): Promise<void> => {
```
[the remark this answers is not in the document]
Comment (puneet-khushwani-eth) [reply test_integration/01_init/02_post_init.test.ts]: done

File: test_integration/03_stake_and_mint/01_single_staker_single_facilitator.test.ts [hunk 6]
```diff
@@ -522,8 +389,8 @@ describe('facilitator start', async (): Promise<void> => {
           } else {
             throw new Error(
               `Message status for source in db is ${messageInDb!.sourceStatus} but in `
-              + `eip20gateway is ${eip20GatewayMessageStatus} and Message status for target in db is `
-              + `${messageInDb!.targetStatus} but got ${eip20CoGatewayMessageStatus}`,
+                + `eip20gateway is ${eip20GatewayMessageStatus} and Message status for target in db is `
```
[the remark this answers is not in the document]
Comment (puneet-khushwani-eth) [reply test_integration/03_stake_and_mint/01_single_staker_single_facilitator.test.ts]: done

File: test_integration/faciltiator_start.test.ts [hunk 7]
```diff
@@ -0,0 +1,559 @@
...
+//       clearInterval(progressMintingInterval);
+//       throw err;
+//     });
+//   });
+//
+//   after(async (): Promise<void> => {
+//     execSync(facilitatorKill, { stdio: outputOptions, env: process.env });
+//     fs.removeSync(Directory.getFacilitatorConfigPath(auxChainId.toString()));
+//   });
+// });
```
[the remark this answers is not in the document]
Comment (puneet-khushwani-eth) [reply test_integration/faciltiator_start.test.ts]: done

File: test_integration/05_stop/01_stop.test.ts [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import { execSync } from 'child_process';
+import * as path from 'path';
+import fs from 'fs-extra';
+import Directory from '../../src/Directory';
+import * as Constants from '../Constants.json';
+
+const facilitatorKill = path.join(__dirname, '../kill_facilitator_process.sh');
+
+describe('facilitator stop', async (): Promise<void> => {
```
[the remark this answers is not in the document]
Comment (puneet-khushwani-eth) [reply test_integration/05_stop/01_stop.test.ts]: done

File: test_integration/03_stake_and_mint/01_single_staker_single_facilitator.test.ts [hunk 9]
```diff
@@ -1,138 +1,64 @@
...
   let simpleTokenInstance: EIP20Token;
   let stakerAccount: Account;
-  const stakerOSTBalance = '20000';
-  const workerOSTBalance = new BigNumber(500);
-  const amountTobeFundedOnOrigin = new BigNumber(1);
-  const amountTobeFundedOnAuxiliary = new BigNumber(1);
-  const testDuration = 3;
-  const interval = 3000;
-  const workerExpirationHeight = '100000000000';
+  let messageHash: string | undefined;
```
[the remark this answers is not in the document]
Comment (puneet-khushwani-eth) [reply test_integration/03_stake_and_mint/01_single_staker_single_facilitator.test.ts]: I had tried that during development but it doesn't work as  we have messageHash? in Class definition.

File: test_integration/04_redeem_and_unstake/01_single_redeemer_single_facilitator.test.ts [hunk 10]
```diff
@@ -0,0 +1,421 @@
...
+      const gateway: EIP20Gateway = utils.getEIP20GatewayInstance();
+
+      assertMessageInterval = setInterval(async (): Promise<void> => {
+        const redeemRequestDb: MessageTransferRequest | null = await utils.getMessageTransferRequest(
+          preGeneratedRedeemRequestHash,
+        );
+
+        if (redeemRequestDb!.messageHash) {
+          ({ messageHash } = redeemRequestDb!);
+          expectedMessage = new Message(
```
[the remark this answers is not in the document]
Comment (puneet-khushwani-eth) [reply test_integration/04_redeem_and_unstake/01_single_redeemer_single_facilitator.test.ts]: requestRedeem which was the biggest of them, broke it into logical sub parts

File: test_integration/01_facilitator_init/02_post_init.test.ts [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+import MosaicConfig from '@openst/mosaic-chains/lib/src/Config/MosaicConfig';
+import { Organization } from '@openst/mosaic-contracts/dist/interacts/Organization';
+import { TransactionReceipt } from 'web3-core';
+import { FacilitatorConfig } from '../../src/Config/Config';
+import Utils from '../Utils';
+import * as Constants from '../Constants.json';
+import assert from '../../test/test_utils/assert';
+import SharedStorage from '../SharedStorage';
+
+describe('facilitator post init', async (): Promise<void> => {
```
[the remark this answers is not in the document]
Comment (puneet-khushwani-eth) [reply test_integration/01_facilitator_init/02_post_init.test.ts]: broke this into multiple test files for clear logical seperation

**Pull request merged**